docs/07-lifecycle.md

```diff
@@ -14,5 +60,7 @@ Apps are installed with `apps install
 - Also Uninstall/Enable/Disable per App
 
 ## Uninstall
+Sysadmin can uninstall an app from Marketplace or using the uninstall slash command. On uninstallation appropriate bot and an OAuth app are deleted, `OnUninstall` call is sent to the app as well. Worth mentioning that the current implementation is not deleting the user data.
+
 ## Upgrade/downgrade consideration
```

Comment:
move (and expand) the upgrade discussion here? calling out not tackling downgrade?

Reply:
I think this should be a separate ticket in Jira. The `master` implementation now sends `OnVersionChanged` message, and I propose that we only include the Bot credentials in it; Bot is sufficient to migrate the KV values stored; and to send users messages if they need to do something requiring user tokens.

I am 1/5 that we should scrape the "handling of upgrades/downgrades" altogether until we have a more solid use-case.

Reply:
I'm with @levb here. Down/upgrade are a hard topic and we don't know good use cases right now.